### PR TITLE
This adds in a function allowing an e2 to set a props position around itself, in a cvar set range.

### PR DESCRIPTION
**Pull request by TechnichalDoge**
This allows e2s that need to move props around itself to assemble itself and stuff to do so. The only abuse i see is if someone programmed an e2 to fly with this, but then again, is it abuse?

### CODE REVIEW

File: lua/entities/gmod_wire_expression2/core/custom/cfc_entity.lua [hunk 1]
```diff
@@ -52,4 +52,20 @@ e2function void entity:ejectPodTo(vector pos)
...
+    local lengthPos = posToLoc[1]^2 + posToLoc[2]^2 + posToLoc[3]^2 -- this uses 3d pythagorean therom, but doesnt square root the end result, as that would just waste time when we could square the convar, which would use less power to calculate
+
+    local entB4PosLoc = this:GetPos() - self:GetPos() -- Localizes the target entity
+    local entLength = entB4PosLoc[1]^2 + entB4PosLoc[2]^2 + entB4PosLoc[3]^2 -- we dont need to do the square root on this, itll just waste time and resources
+
+    if lengthPos >= CFCMaxFairSetDist^2 then return -1 end -- Debug output -1 if the wanted pos is out of bounds and end
+    if entLength >= CFCMaxFairSetDist^2 then return -2 end -- Debug output -2 if the target entity pos is out of bounds and end
+
+    return 1 --passed all tests, debug 1 for success and move on
+    this:setPos(target) -- set the pos and end it
```
Comment (PLally) [inline lua/entities/gmod_wire_expression2/core/custom/cfc_entity.lua]: isnt this code is unreachable, you return before setting the position

Comment (TechnichalDoge) [reply lua/entities/gmod_wire_expression2/core/custom/cfc_entity.lua]: Fixed

File: lua/entities/gmod_wire_expression2/core/custom/cfc_entity.lua [hunk 2]
```diff
@@ -52,4 +52,20 @@ e2function void entity:ejectPodTo(vector pos)
 
     driver:ExitVehicle()
     driver:SetPos(clampedPos)
-end
\ No newline at end of file
+end
+
+e2function number entity:setFairPos(vector target)
+    if this:IsPlayer() then return -3 end -- Debug output -3 if the target is a player
+    local CFCMaxFairSetDist = GetConVar("CFC_Lib_MaxFairSetDistance"):GetFloat()
```
Comment (PLally) [inline lua/entities/gmod_wire_expression2/core/custom/cfc_entity.lua]: maybe just `maxDistance` the cfc part is unnecessary as its a a local variable

Comment (TechnichalDoge) [reply lua/entities/gmod_wire_expression2/core/custom/cfc_entity.lua]: Agreed

**Pull request closed without merging**